File: BOJ/SeokhyeonHeo/애너그램/BOJ_6996.py
# -*- coding: utf-8 -*-

N = int(input())

def anagram(A, B):
    
    A_hashTable = {}
    B_hashTable = {}
    
    for i in A:
        if A_hashTable.get(i):
            A_hashTable[i] += 1
        else:
            A_hashTable[i] = 1

    for j in B:
        if B_hashTable.get(j):
            B_hashTable[j] += 1
        else:
            B_hashTable[j] = 1
            
    if A_hashTable == B_hashTable:
        return print(A+' & '+B+' are anagrams.')
    else:
        return print(A+' & '+B+' are NOT anagrams.')


for i in range(N):
    A, B = input().split()
    anagram(A, B)

File: BOJ/ParkJihyeon/문자열/BOJ_6996.java
package ParkJihyeon.문자열;

import java.io.BufferedReader;
import java.io.IOException;
import java.io.InputStreamReader;
import java.util.Arrays;
import java.util.StringTokenizer;

public class BOJ_6996 {
    static void anagram(String first, String second) {
        char[] charArr = first.toCharArray();
        char[] charArr2 = second.toCharArray();

        Arrays.sort(charArr);
        Arrays.sort(charArr2);
        System.out.println(charArr);
        System.out.println(charArr2);
        if (Arrays.equals(charArr, charArr2)) {
            System.out.println(first + " & " + second + " are anagrams.");
        } else {
            System.out.println(first + " & " + second + " are NOT anagrams.");
        }
    }
    public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        int N = Integer.parseInt(br.readLine());

        for (int i = 0; i < N; i++) {
           // StringTokenizer st = new StringTokenizer(br.readLine());
            String[] str = br.readLine().split(" " );
            String first = str[0];
            String second = str[1];
            anagram(first, second);
        }
    }
}

